packages/vm/src/errors.rs

```diff
@@ -117,7 +117,24 @@ impl From<wasmer_runtime_core::error::ResolveError> for VmError {
 
 impl From<wasmer_runtime_core::error::RuntimeError> for VmError {
     fn from(original: wasmer_runtime_core::error::RuntimeError) -> Self {
-        make_runtime_err(format!("Wasmer runtime error: {:?}", original))
+        use wasmer_runtime_core::error::{InvokeError, RuntimeError};
+
+        fn runtime_error(err: RuntimeError) -> VmError {
+            make_runtime_err(format!("Wasmer runtime error: {:?}", err))
+        }
+
+        match original {
+            // `InvokeError::FailedWithNoError` happens when running out of gas in singlepass v0.17
+            // but it's supposed to indicate bugs in Wasmer...
+            // https://github.com/wasmerio/wasmer/issues/1452
+            RuntimeError::InvokeError(InvokeError::FailedWithNoError) => VmError::GasDepletion,
+            // This variant contains the error we return from imports.
```

Comment:
So this is used to unwrap another error... interesting check. It would be good to link to a reference here, or better test this works. Otherwise this is just some magic downcast that may or may not have an effect.

Reply:
This unwraps the error type that we return from the imports (remember the earlier change where imports now return `Result`s?). I'll add a test that makes sure the error is unwrapped, which should make sure that this matches the type in the imports declaration.

Reply:
Ah, cool explanation. Nice one.

Please just add a test that one of these import errors passes through correctly here.

Reply:
aaand, done :smile: